esp-hal/src/analog/dac.rs

```diff
@@ -12,28 +12,29 @@
 //! ## Example
 //!
 //! ```no_run
-//! let io = Io::new(peripherals.GPIO, peripherals.IO_MUX);
-//! let gpio25 = io.pins.gpio25;
-//! let gpio26 = io.pins.gpio26;
+#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/doc-helper/before"))]
```

Comment:
See https://doc.rust-lang.org/std/macro.include_str.html - assuming `/` as the path separator might cause subtle problems on Windows

ideally you could create a macro which builds the path, then we could have something like `#![doc = include_str!(before_snippet!())]` here and it would be easy to have two versions of that macro (one for Windows, one for all other OS)

Reply:
Good idea! I updated it and added macros